### PR TITLE
nix: improvements

### PR DESCRIPTION
**Pull request by ratsclub**
# Motivation

The idea of this PR is to improve the [Nix](https://nixos.org) support of our application.

- [x] GitHub Action to check if our package is building
- [x] An overlay with the newest version

### CODE REVIEW

**Comment by kmaasrud** [comment]
❄️❄️

**Comment by ratsclub** [comment]
@kmaasrud let me know what you think about this. :smile:

File: flake.nix [hunk 1]
```diff
@@ -7,43 +7,72 @@
...
-          pkgs.mkShell { nativeBuildInputs = with pkgs; [ rust-toolchain ]; };
+          pkgs.mkShell {
+            nativeBuildInputs = with pkgs; [
+              # rust
+              cargo
+              clippy
+              rust-analyzer
+              rustc
+
+              # misc
```
Comment (kmaasrud) [inline flake.nix:66]: Let's just add `fswatch` (needed for live-updating docs) here as well, so that we don't need two dev-shells.

**Review by kmaasrud — COMMENTED** [review]
LGTM, I think I'll just clean up the `docs` shell, so we can use the one dev-shell for everything.

**Comment by kmaasrud** [comment]
@ratsclub allright, I removed the `docs` shell and ran `nixfmt` to format the nix files, so I think we're all good now 😎

**Review by kmaasrud — APPROVED** [review]

**Comment by kmaasrud** [comment]
Pretty sure `website.nix` works now!

**Review by kmaasrud — APPROVED** [review]
All good man, let's merge this mf 🤠

**Pull request merged**